src/scheduler.cpp

```diff
@@ -196,6 +196,9 @@ void SingleThreadedSchedulerClient::AddToProcessQueue(std::function<void ()> fun
         m_callbacks_pending.emplace_back(std::move(func));
     }
     MaybeScheduleProcessQueue();
+    if (!m_pscheduler->AreThreadsServicingQueue()) {
+        EmptyQueue();
```

Comment:
In commit "Handle no threads serving in MaybeScheduleProcessQueue" (201eebf13b2723d7fe821d7222600fa04fadc0b0)

Commit message subject is out of date since `MaybeScheduleProcessQueue` is no longer changing.

Also I'm surprised by "SyncWithValidationInterfaceQueue is called after
scheduler thread exists, which happen at shutdown" since I would hope wallets are getting unloaded before the scheduler stops, but maybe this is not the case currently. Probably beyond the scope of this PR to address, though

Reply:
> since I would hope wallets are getting unloaded before the scheduler stops

Not currently the case.

> Probably beyond the scope of this PR to address, though

Agree, I've tried and it caused some other failures.

> Commit message subject is out of date since MaybeScheduleProcessQueue is no longer changing.

Ops need to update.